packages/core/src/controllers/subscriber.ts

```diff
@@ -382,13 +390,14 @@ export class Subscriber extends ISubscriber {
 
   private async batchSubscribe(subscriptions: SubscriberTypes.Params[]) {
     if (!subscriptions.length) return;
+
     const result = (await this.rpcBatchSubscribe(subscriptions)) as string[];
     if (!isValidArray(result)) return;
+
     this.onBatchSubscribe(result.map((id, i) => ({ ...subscriptions[i], id })));
   }
 
   private async onConnect() {
-    if (this.restartInProgress) return;
     await this.restart();
```

Comment:
What happens if there is there is already one in progress now?

Reply:
It will restart the transport. Logic is, and confirmed that it happens in the wild
- open socket
- attempt to subscribe
- disconnect received while subscribe is still in progress
- retry all again